### PR TITLE
diagnose subdomain e_tot and h_tot 

### PR DESCRIPTION
**Pull request by yairchn**
this PR is a peel off PR for changing subdomain variables to `e_tot` by adding the following:

* adds `e_tot` and `h_tot` in the updraft, env and bulk.
* updates `e_tot`,`h_tot` in update_aux.
* decompose the environment based on `h_tot` and not  `e_tot`.

### CODE REVIEW

File: src/update_aux.jl [hunk 1]
```diff
@@ -53,6 +63,11 @@ function update_aux!(edmf::EDMFModel, grid::Grid, state::State, surf::SurfaceBas
                     aux_up[i].θ_liq_ice[k] = θ_surf
                     aux_up[i].q_tot[k] = q_surf
                     aux_up[i].area[k] = a_surf
+                    thermo_args = if edmf.moisture_model isa EquilibriumMoisture
+                        ()
+                    elseif edmf.moisture_model isa NonEquilibriumMoisture
+                        (aux_up[i].q_liq[k], aux_up[i].q_ice[k])
+                    end
```
Comment (charleskawczynski) [inline src/update_aux.jl:70]: What's this being added for?

File: driver/generate_namelist.jl [hunk 2]
```diff
@@ -106,7 +106,7 @@ function default_namelist(
     namelist_defaults["turbulence"]["EDMF_PrognosticTKE"]["entrainment_smin_tke_coeff"] = 0.3
     namelist_defaults["turbulence"]["EDMF_PrognosticTKE"]["updraft_mixing_frac"] = 0.25
     namelist_defaults["turbulence"]["EDMF_PrognosticTKE"]["area_limiter_scale"] = 10.0
-    namelist_defaults["turbulence"]["EDMF_PrognosticTKE"]["area_limiter_power"] = 3.0
+    namelist_defaults["turbulence"]["EDMF_PrognosticTKE"]["area_limiter_power"] = 4.0
```
Comment (charleskawczynski) [inline driver/generate_namelist.jl:109]: Why is this changing?

Comment (yairchn) [reply driver/generate_namelist.jl:109]: it cleans up the spikes in Soares

File: src/update_aux.jl (hunk 1 above)
Comment (yairchn) [reply src/update_aux.jl]: leftover something I tried , I cleared it

**Review by charleskawczynski — APPROVED** [review]

**Comment by yairchn** [comment]
Bors r+

File: src/thermodynamics.jl [hunk 3]
```diff
@@ -39,6 +39,14 @@ function anelastic_total_enthalpy(param_set::APS, e_tot::FT, p::FT, ρ::FT) wher
     return e_tot + p / ρ
 end
 
+function anelastic_total_enthalpy(h::FT, e_pot::FT, e_kin::FT) where {FT}
```
Comment (tapios) [inline src/thermodynamics.jl:42]: Please be careful with the term anelastic here. This isn't anelastic. In the anelastic approximation, the total enthalpy would involved buoyancy (and generally buoyancy integrals).

File: src/thermodynamics.jl [hunk 4]
```diff
@@ -39,6 +39,14 @@ function anelastic_total_enthalpy(param_set::APS, e_tot::FT, p::FT, ρ::FT) wher
...
+    return h + e_pot + e_kin
+end
+
 function kinetic_energy(u::FT, v::FT, w::FT) where {FT}
     return FT(0.5) * (u^2 + v^2 + w^2)
 end
+
+function specific_enthalpy(h_tot::FT, e_kin::FT, e_pot::FT) where {FT}
```
Comment (tapios) [inline src/thermodynamics.jl:50]: I'd just call this enthalpy. Specific just refers to quantities referenced to unit mass, which all our thermodynamic quantities are. So strictly, it's "specific total enthalpy" etc. Adding specific here and not for total enthalpy just confuses. If you want to make it clearer, you can call it "thermodynamic_enthalpy" or something like it.

File: src/update_aux.jl [hunk 5]
```diff
@@ -66,8 +76,12 @@ function update_aux!(edmf::EDMFModel, grid::Grid, state::State, surf::SurfaceBas
                     aux_up[i].θ_liq_ice[k] = aux_gm.θ_liq_ice[k]
                     aux_up[i].q_tot[k] = aux_gm.q_tot[k]
                     aux_up[i].area[k] = 0
+                    aux_up[i].e_kin[k] = aux_gm.e_kin[k]
                 end
             end
+            ts_up_i = thermo_state_pθq(param_set, p_c[k], aux_up[i].θ_liq_ice[k], aux_up[i].q_tot[k])
```
Comment (charleskawczynski) [inline src/update_aux.jl:82]: Not a blocker, but this technically assumes equilibrium thermodynamics, correct?

**Review by charleskawczynski — APPROVED** [review]
Code changes LGTM

File: src/update_aux.jl [hunk 6]
```diff
@@ -37,13 +37,22 @@ function update_aux!(edmf::EDMFModel, grid::Grid, state::State, surf::SurfaceBas
     ts_gm = center_aux_grid_mean(state).ts
     ts_env = center_aux_environment(state).ts
 
+    prog_gm_u = grid_mean_u(state)
+    prog_gm_v = grid_mean_v(state)
+    Ic = CCO.InterpolateF2C()
     #####
     ##### center variables
     #####
+    @. aux_en.e_kin = kinetic_energy(prog_gm_u, prog_gm_v, Ic(FT(0) + aux_en_f.w))
```
[the remark this answers is not in the document]
Comment (yairchn) [reply src/update_aux.jl:46]: ask the folks that coded ClimaCore ....

**Pull request merged**